inc/lift/Header.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <string>
+#include <string_view>
+
+namespace lift {
+
+class Header {
+public:
+    /**
+     * Creates an owned header.
+     * @param name The name of the header.
+     * @param value The value of the header.
+     */
+    explicit Header(
+        std::string_view name,
+        std::string_view value);
+
+    /**
+     * @return The entire header, e.g. "Connection: Keep-Alive"
+     */
+    [[nodiscard]] auto HeaderFull() const -> const std::string& { return m_header; }
+    [[nodiscard]] auto HeaderFull() -> std::string& {return m_header; }
+
+    /**
+     * @return The header's name.
+     */
+    [[nodiscard]] auto Name() const -> std::string_view { return m_name; }
+
+    /**
+     * @return True if this header has a value.
+     */
+    [[nodiscard]] auto HasValue() const -> bool { return !m_value.empty(); }
```

Comment:
I think there are some headers where we want an empty string. Maybe this should check if the string_view is engaged?